cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -621,6 +623,19 @@ private static void configureCruiseControlMetrics(Kafka kafkaAssembly, KafkaClus
         }
     }
 
+    /**
+     * Early resources validation to avoid triggering any pod operation with invalid configuration.
+     * 
+     * @param resources Resources configuration.
+     */
+    private static void validateComputeResources(ResourceRequirements resources) {
+        Set<String> errors = ModelUtils.validateCpuResources(resources, "kafka");
```

Comment:
Maybe instead of `kafka` you can use the YAML path? E.g. `.spec.kafka.resources` or `.spec.kafka.zookeeper`?